### PR TITLE
Tidy up descriptor offset relocation changes

### PR DESCRIPTION
**Pull request by trenouf**
* Moved emitRelocationConstant from Internal to
BuilderBase::CreateRelocationConstant
* Also subsumed the reformatting of the same change from #594.

Change-Id: Ifc3e332af22e0cdac138517cf363622caaca11f3

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 84a3d6a848087563674bf627c213a172c3df43f9
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev b5e6dde2a0aa5dd2ce6d794e58d89ef715570f2d</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103142/103142 (100.0%)</li>
<li>Failed: 0/103142 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 89041/89042 (100.0%)</li>
<li><details><summary>Failed: 1/89042 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.subgroups.clustered.compute.subgroupclusteredadd_uint16_t
Stack trace: Timeout or crahsed
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit f8ab725c2847f3d451c1b680934299237d30c31b
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev b5e6dde2a0aa5dd2ce6d794e58d89ef715570f2d</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103142/103142 (100.0%)</li>
<li>Failed: 0/103142 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/util/llpcElfWriter.cpp [hunk 1]
```diff
@@ -1143,15 +1143,15 @@ bool getDescriptorOffsetRelocationValue(Context *context, RelocationEntry relocE
   if (isGraphicsPipeline) {
     auto pPipelineInfo = reinterpret_cast<const GraphicsPipelineBuildInfo *>(context->getPipelineBuildInfo());
```
Comment (amdrexu) [inline llpc/util/llpcElfWriter.cpp:1144]: I see you have made a change to correct the coding style issue. And you reverted it because of the backward compatibility problem. I think the function-internal naming is still safe to be corrected like pPipelineInfo -> pipelineInfo. It could be another PR in order to not distract the audience from this one.

File: llpc/util/llpcElfWriter.h [hunk 2]
```diff
@@ -97,7 +97,7 @@ template <class Elf> class ElfWriter {
 
   Result getSectionDataBySectionIndex(unsigned secIdx, const SectionBuffer **ppSectionData) const;
 
-  Result getSectionData(const char* pName, const void** ppData, size_t* pDataLength) const;
+  Result getSectionData(const char *pName, const void **ppData, size_t *pDataLength) const;
```
Comment (amdrexu) [inline llpc/util/llpcElfWriter.h:100]: The same as above

File: llpc/util/llpcElfWriter.cpp (hunk 1 above)
Comment (trenouf) [reply llpc/util/llpcElfWriter.cpp:1144]: Thanks Rex. Shall we leave that to another PR? Are you able to approve this one please?

Comment (amdrexu) [reply llpc/util/llpcElfWriter.cpp:1144]: I'm OK with making another PR to correct this. I can approve this if no one has additional comments.
